app/src/main/java/com/cornellappdev/android/eatery/components/WaitTimesMarkerView.java

```diff
@@ -51,19 +52,20 @@ public void updateMarkerLabel(Entry e, Swipe s) {
         float eX = e.getX();
         int t = (int)((eX + 6) % 12);
         t = t == 0 ? 12 : t;
-        String time = t + (eX <= 5 || eX >= 18 ? "a" : "p") + ": ";
-        String waitTime = s.waitTimeLow + "-" + s.waitTimeHigh + "m";
+        boolean hasNoData = s.waitTimeLow == 0 && s.waitTimeHigh == 0;
+        String time = (LocalTime.now().getHour() - 6 == eX ? "Now" : (t + (eX <= 5 || eX >= 18 ? "a" : "p"))) + ": ";
+        String waitTime = hasNoData ? "?" : (s.waitTimeLow + "-" + s.waitTimeHigh + "m");
```

Comment:
yikes this is really hard to read 😬 is it possible to use a `String.format()` or will it make the code less readable?

Reply:
I broke it up into helper functions and added additional comments.